### PR TITLE
refactor(tooltip, popover, input-date-picker): `onToggleOpenCloseComponent` replaces connect logic

### PR DESCRIPTION
**Pull request by Elijbet**
**Related Issue:** #6018

## Summary
Use the new `onToggleOpenCloseComponent` pattern to replace both `connectOpenCloseComponent` and `disconnectOpenCloseComponent` on `tooltip`, `popover`, and `input-date-picker`. This new pattern is intended to simplify the overall eventing logic.

### CODE REVIEW

**Review by driskull — APPROVED** [review]
These code changes look good to me 👍

File: packages/calcite-components/src/components/popover/popover.tsx [hunk 1]
```diff
@@ -300,9 +298,9 @@ export class Popover
     this.setFilteredPlacements();
     connectLocalized(this);
     connectMessages(this);
-    connectOpenCloseComponent(this);
     this.setUpReferenceElement(this.hasLoaded);
     connectFocusTrap(this);
+    onToggleOpenCloseComponent(this);
```
Comment (jcfranco) [inline packages/calcite-components/src/components/popover/popover.tsx:303]: This needs to be called only when `open = true`, no?

**Review by jcfranco — APPROVED** [review]
🎉

This should be good to merge once comment(s) are addressed.

File: packages/calcite-components/src/components/popover/popover.tsx (hunk 1 above)
Comment (Elijbet) [reply packages/calcite-components/src/components/popover/popover.tsx:303]: yes!

**Pull request merged**